### ISSUE TITLE
Test explorer should use 'dotnet build' instead of 'msbuild /t:Build' to build the test project for test discovery

### ISSUE THREAD

**Issue by Martin521** [ionide/ionide-vscode-fsharp#1967]
### Describe the bug
When working on the dotnet/fsharp repository in a dev container, the test explorer test discovery fails with a build error.
Reason is
- The test explorer uses `msbuild /t:Build`
- To build the compiler, you must use their build scripts, or use `dotnet build`, or have `BUILDING_USING_DOTNET` set.
(This is true once https://github.com/dotnet/fsharp/pull/16335 is merged.)

### Steps to reproduce
Open the dotnet/fsharp directory in VS Code Dev Containers.
Wait for the (misleading) error message `Couldn't build test projects. Make sure you can build projects with 'dotnet build'`.

## Machine info
 - OS: Windows 11 with WSL2/docker and VS Code Dev Containers
 - .NET SDK version: 8.0.100-rc.1.23463.5 in dev container
 - Ionide version: 7.16.1

### Additional context
As `dotnet test` is used for discovery and running of tests, I assume `dotnet build` is ok for building. Or is there a specific reason to use 'msbuild' directly? It would be nice to have test discovery work out of the box when opening dotnet/fsharp in a dev container.
In any case, the error message should be changed to say `... Make sure you can build projects with 'msbuild /t:Build'`.

**Comment by TheAngryByrd** [comment]
cc @farlee2121

**Comment by farlee2121** [comment]
The test explorer uses msbuild because
- msbuild returns a signal, so we don't have to parse to tell if the build failed. Dotnet build returns command success even if the build fails.
- consistency with other Ionide commands, which use msbuild

As for the error message, the vast majority of the time `dotnet build` is going to boil down to the same result as msbuild
but is much more likely to be in the user's path and thus easy for them to double check.
I don't know for sure, but I'd guess dotnet build is a light wrapper around msbuild. 

Considering tests were discovered for the fsharp repo in #1946, there might be something else going on here.

**Comment by baronfel** [comment]
> dotnet build returns command success even if the build fails

This is surprising to me - we're getting the exit code from MSBuild [here](https://github.com/dotnet/sdk/blob/5433bc4745265d7c16611f2b389b780e834eacc0/src/Cli/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs#L83-L111) and should be flowing that up. Can you log an issue to dotnet/sdk with a repro?

**Comment by farlee2121** [comment]
Maybe I need to revisit this and look more closely at the exit code, but I remember stumbling into this during my work.
My notes say build failure didn't result in an error code and I linked to this existing issue

https://github.com/dotnet/sdk/issues/8481#issuecomment-314175413

**Comment by Martin521** [comment]
This is working now, so I am closing the issue.
